### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by eduazocar**
### Contribution description
This PR is used only to write the Contributing guidelines and other written documents about the repository, it's necessary for know how all the people could help with the project and use it. The main idea is detect all possible's issues and fix it from other users. more information is writing in [CONTIBUTING.md](https://github.com/Mesh4all/m4ag-firmware/blob/feature/CONTRIBUTING/CONTRIBUTING.md)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+## Getting Started 
+
+---
+
+if you are starting work with mesh4all, you can read our documentation, and you will know all supported boards, and resources.
+
+First Steps:
+1. Clone the repository.
+2. Update submodules.
+3. Install RIOT-OS.
```
Comment (luisan00) [inline CONTRIBUTING.md:21]: Already it's  instaled as submodule so this step is unnecessary

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# Contributing to Mesh4all - Ag (Agriculture Firmware)
+
+Welcome to contribute to Mesh4all-ag, all ideas are welcome, feel free to share your requirements, comments, bugs or problems.
+Remember that the main idea is to connect everyone and everything around us.
+
+Check all the following sections according to the requirement you need.
+
+- ### <a href="#titlegs">Getting Started</a>
+- ### <a href="#titlebrfr"> Bug Reports and Feature Request</a>
+- ### <a href="#titlecc"> Contributing Code</a>
```
Comment (luisan00) [inline CONTRIBUTING.md:10]: Don't use heading marks in this list

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+First Steps:
+1. Clonning
+2. Update submodules.
+
+<br>
+To start with m4ag-firmware, we need to clone the repository:
+
+<br>
+
+### **1. Clonning**
```
Comment (luisan00) [inline CONTRIBUTING.md:29]: Don't use bold syntax here, h3 heading is enough

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+#### Via HTTPS: 
+```sh
+    git clone https://github.com/Mesh4all/m4ag-firmware.git
+```
+
+*Note:* We recommend cloning through SSH, due to external dependencies of some module
+
+<br>
+
+### **2. Updating Submodules**
```
Comment (luisan00) [inline CONTRIBUTING.md:44]: Same here

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+--- 
+
+if you think that your work will be a growth for the project, integrate in the repository m4ag-firmware, we recommend: 
+
+1. Fork the m4ag-firmware git repository.
+2. Create your contribution branch.
+3. Give a git push to you branch.
+4. Generate a [Pull Request](https://github.com/Mesh4all/m4ag-firmware/compare) (if you can't see your branch, try repeat the third step).
+
+### Thank you very much for your future contribution, we are now connected together in this great mesh around the world!!!.
```
Comment (luisan00) [inline CONTRIBUTING.md:81]: Don't use here H3, I think it's better bold or none, simple text

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    git clone https://github.com/Mesh4all/m4ag-firmware.git
+```
+
+*Note:* We recommend cloning through SSH, due to external dependencies of some module
+
+<br>
+
+### **2. Updating Submodules**
+
+Our firmware is based on RIOT-OS, so you have to apply updates from the RIOT repository within the m4ag firmware using the following command lines with git.
```
Comment (luisan00) [inline CONTRIBUTING.md:46]: link to RIOT-OS repo in the first mention

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+- ### <a href="#titlebrfr"> Bug Reports and Feature Request</a>
+- ### <a href="#titlecc"> Contributing Code</a>
+
+<br>
+
+<h2 id="titlegs"> Getting Started </h2>
+
+---
+
+if you are starting work with mesh4all, you can read our documentation, and you will know all supported boards, and resources.
```
Comment (luisan00) [inline CONTRIBUTING.md:18]: .. starting **to** work...

**Review by luisan00 — CHANGES_REQUESTED** [review]

**Comment by luisan00** [comment]
Now looks good @eduazocar , please do squash with all commits before to merge and set the commit msg to `docs: add contributing guidelines` or something like that but, important precede `docs:`
Do: `git rebase -i main`

**Comment by eduazocar** [comment]
Don't pay attention to all the attempts of merge xD, I forgot when i changed the README.md for some tests

**Comment by eduazocar** [comment]
Don't pay attention to all the attempts of merge, I forgot when i changed the README.md for some test

**Comment by luisan00** [comment]
Looks good @eduazocar, go abead

**Review by luisan00 — APPROVED** [review]

**Pull request merged**